File: LeetCode/56/main.cpp
class Solution {
public:
  vector<vector<int>> merge(vector<vector<int>> &arr) {
    int n = arr.size();
    vector<vector<int>> ans;

    sort(arr.begin(), arr.end());
    ans.emplace_back(arr[0]);

    for (int i = 1; i < n; ++i) {
      vector<int> prev = ans[ans.size() - 1];
      vector<int> cur = arr[i];

      if (cur[0] >= prev[0] && cur[0] <= prev[1]) {
        int low = min(prev[0], cur[0]);
        int high = max(prev[1], cur[1]);
        ans[ans.size() - 1] = {low, high};
      } else {
        ans.emplace_back(arr[i]);
      }
    }

    return ans;
  }
};

File: LeetCode/56/main.go
package main

import (
	"cmp"
	"slices"
)

func merge(arr [][]int) [][]int {

	n := len(arr)

	slices.SortFunc(arr, func(a, b []int) int {
		return cmp.Compare(a[0], b[0])
	})

	ans := [][]int{arr[0]}

	for i := 1; i < n; i++ {
		prev, cur := ans[len(ans)-1], arr[i]

		if cur[0] >= prev[0] && cur[0] <= prev[1] {
			mn, mx := min(cur[0], prev[0]), max(cur[1], prev[1])
			ans[len(ans)-1] = []int{mn, mx}
		} else {
			ans = append(ans, arr[i])
		}
	}

	return ans

}
